### ISSUE TITLE
[5.4.0-rc.0] Administration -> Apps does not work

### ISSUE THREAD

**Issue by paulchen** [RocketChat/Rocket.Chat#27346]
### Description:

After navigating to the "Administration" page, clicking the menu item "Apps" does not work.

### Steps to reproduce:

1. Click on the kebap menu above the channel list and select "Administration".
2. Click on the menu item "Apps".

### Expected behavior:

The page `/admin/marketplace/all/list` is opened.

### Actual behavior:

Nothing happens.

Clicking on "Marketplace" in the kebab menu works, as well as explicitely navigating to the Apps page by entering the URL into the address bar.

### Server Setup Information:

- Version of Rocket.Chat Server: 5.4.0-rc.0
- Deployment Method: Docker
- Number of Running Instances: 1
- MongoDB Version: 5.0.14 / wiredTiger (oplog Enabled)

### Client Setup Information

- Desktop App or Browser Version:
  - Version 107.0.5304.110 (Offizieller Build) Arch Linux (64-Bit)
  - Mozilla Firefox for Arch Linux 107.0 (64-bit)
- Operating System: Arch Linux

### Relevant logs:

Upon clicking the menu item "Apps", the following error is logged in the Browser console:

```
95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483 Uncaught Error: redirect needs to be done in sync
    at c (95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483:1355)
    at s.triggersEnter (95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1580:3634179)
    at h.runTriggers (95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483:1271)
    at o._actionHandle (95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483:2874)
    at 95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483:14766
    at o (95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483:14186)
    at 95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483:14773
    at o (95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483:14186)
    at 95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483:14773
    at o (95bbc042fcc189879df953f0c42d5e017736ef36.js?meteor_js_resource=true:1483:14186)
```

**Comment by casalsgh** [comment]
Hi @paulchen thanks for the report here. 
We tried to replicate the issue, but couldn't, at least on the latest version of 5.4.0. Below gif is our view and all seems ok. 
![marketplace_not_loading](https://user-images.githubusercontent.com/83978645/204003170-ea04148b-3092-454f-b204-d26b6bc74ef4.gif)

Can you try again with latest version and see if all good? if not, would you mind sharing with us in more details the steps to reproduce? 
Many thanks!

**Comment by casalsgh** [comment]
Interesting. Ok, when you try with a fresh set let us know. 
Thanks for feedback and confirming versions and steps to reproduce.

**Comment by paulchen** [comment]
I can reproduce this in a fresh installation of Rocket.Chat using Docker on Arch Linux:

1. Start MongoDB: `docker run --name mongo_test mongo:5.0.14 mongod --oplogSize 128 --replSet rs0 --storageEngine=wiredTiger`
2. Initialize the MongoDB replica set: `docker exec -it mongo_test mongo localhost/rocketchat --eval "rs.initiate({ _id: 'rs0', members: [ { _id: 0, host: 'localhost:27017' } ]})"`
3. Run Rocket.Chat: `docker run --name rocketchat_test -e MONGO_OPLOG_URL=mongodb://mongo:27017/local?directConnection=true -e MONGO_URL=mongodb://mongo:27017/rocketchat?directConnection=true --link mongo_test:mongo -p 127.0.0.1:3000:3000 rocketchat/rocket.chat:develop`
4. Point the browser to http://localhost:3000 and set up Rocket.Chat as a stand-alone instance.
5. Follow the steps as described in the initial error report.

I recorded my screen when following these steps:

https://user-images.githubusercontent.com/404840/204060129-17657475-4b63-4a7f-8e2c-a5f81ce55611.mp4

Note the German names:
* "Arbeitsbereich" translates to "Workspace"
* "Marktplatz" translates to "Marketplace"
* "Anwendungen" translates to "Apps"

**Comment by casalsgh** [comment]
Checking and will get back

**Comment by graywolf336** [comment]
@paulchen does this happen for on all browsers? Aside from trying it on Arch, we have followed your steps exactly and haven't been able to reproduce it for some odd reason. I'm trying to nail down the differences and why this would be happening to you and not for us.

**Comment by rique223** [comment]
Hey @paulchen. First of all, thanks a lot for finding and reporting this bug. This issue was caused by a problem between a flow router event called triggersEnter, which we use for rerouting in marketplace, and some production build steps, you can read more about the solution for this in the PR mentioned above. Once again, thanks for your input; this should be solved soon once the PR is merged.

**Comment by casalsgh** [comment]
@paulchen fix was merged and will be on the latest version of 5.4.0 release (to be out anytime soon). Thanks for your feedback and engagement here!

**Comment by paulchen** [comment]
@casalsgh @rique223 @graywolf336 Thanks for taking care of this problem!

I just tried out the latest `develop` image from Docker Hub which already contains the fix. I can confirm that the issue is now fixed.
